contracts/marketplace/nft_marketplace.ral

```diff
@@ -61,10 +60,17 @@ Contract NFTMarketPlace(
         let price = nftListing.getPrice()
         let commissionRateOfListing = nftListing.getCommissionRate()
         let listingFeeOfListing = nftListing.getListingFee()
+        let royaltyAmount = nftListing.royaltyAmount()
+        let priceAfterFee = checkListingPriceAfterFee(price, commissionRateOfListing, listingFeeOfListing, royaltyAmount)
```

Comment:
How about passing the `priceAfterFee` to the `NFTListing.buy` function?

If so we don't need to calculate the `priceAfterFee` in the `NFTListing.buy` function, and it's safe because we have checked the caller is the marketplace contract in the `NFTListing.buy` function.

Reply:
Initially, i thought NFTListing should be able to calculate based on its own info: price, commission rate and listing price. But since  the `buy` method needs to be called by marketplace anyways, it should be safe that marketplace calculates this value for it.